pkg/traversal/traversal.go

```diff
@@ -0,0 +1,133 @@
+package traversal
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/openshift/must-gather-clean/pkg/obfuscator"
+	"github.com/openshift/must-gather-clean/pkg/omitter"
+	"github.com/openshift/must-gather-clean/pkg/output"
+)
+
+type FileWalker struct {
+	input       string
+	obfuscators []obfuscator.Obfuscator
+	writer      output.Outputter
+	omitters    []omitter.Omitter
+}
+
+// NewFileWalker returns a FileWalker. It ensures that the input directory exists and is readable.
+func NewFileWalker(inputDir string, writer output.Outputter, obfuscators []obfuscator.Obfuscator, omitters []omitter.Omitter) (*FileWalker, error) {
+	fileInfo, err := os.Stat(inputDir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot access input directory %s: %w", inputDir, err)
+	}
+	if !fileInfo.IsDir() {
+		return nil, fmt.Errorf("input path %s is not a directory", inputDir)
+	}
+	inputPath, err := filepath.Abs(inputDir)
+	if err != nil {
+		return nil, err
+	}
+	return &FileWalker{input: inputPath, obfuscators: obfuscators, writer: writer, omitters: omitters}, nil
+}
+
+// Traverse should be called to start processing the input directory.
+func (w *FileWalker) Traverse() error {
+	return w.processDir(w.input, "")
+}
+
+func (w *FileWalker) processDir(inputDirName string, outputDirName string) error {
+	// list all the entities in the directory
+	entries, err := os.ReadDir(inputDirName)
+	if err != nil {
+		return err
+	}
+	for _, e := range entries {
+		fileInfo, err := e.Info()
+		if err != nil {
+			return err
+		}
+
+		// If a file is a symlink then ignore it
+		if fileInfo.Mode()&os.ModeSymlink != 0 {
+			continue
+		}
+
+		if e.IsDir() {
+			name := e.Name()
+			// obfuscate the directory name
+			for _, o := range w.obfuscators {
+				name = o.FileName(name)
```

Comment:
now that's confusing: 

```
if e.IsDir() {
   name := o.FileName(e.Name())
}
```

I think we don't have to obfuscate the directory names here. It's enough to do it with the files

Reply:
What if the directory name contains a hostname or IP address?

Reply:
does that happen? If yes we should either change the `FileName` method to `FilePath` as it can also match directories, or only match on file names

Reply:
Since it's only obfuscating the name(either directory or file) I don't see how changing it to `FilePath` would be more correct.

Reply:
then let's keep the "FileName" method and not obfuscate the directories

Reply:
👍🏼